### ISSUE TITLE
Datagrid show allow footer extension template that supports auto-alignment with header column sizes

### ISSUE THREAD

**Issue by mattmutt** [vmware/clarity#3647]
## Summary

As the grid provides a header for aligning data columns, naturally there should be a means to placing tabular data in a special set of row(s) underneath the main body in a footer, with the same alignment rules in place.

* What is the change?

A new UI footer extension point - perhaps in the form of a template that allows clients to place content underneath the main body data rows into a fixed row. This data is not part of the grid data, but rather computed or summary values.

* Why should it go in Clarity?

To benefit the consumers of the datagrid who demand an easy way to place column-aligned content in the footer region. They are accustomed to showing prominent values in a common fixed area just under the grid data.

* Does this change impact existing behaviors? If so how?
Changes the UX slightly as there is more content in the footer region. However the pagination still works, just shifted down to look reasonable.

* If this change introduces a new behavior, is this behavior accessible?

Yes, accessibility is not impaired. There may be more content for the footer summary row, but it would comply with source order tab index rules.

## Use case

Common use cases already in shipped products like Excel and spreadsheet apps.

Summary rows: Financial and Scientific grids use many numbers, so having a way to describe totals underneath any column makes sense. Since the client would be responsible for the value generation with their own method, they could easily take into consideration data set, pagination rules. It might even be pre-computed for big data.

Placement of column actions: Some use cases can call for actions applied at the column level. Hence buttons may be useful if the customer UX demands them in a fixed, visible location in the footer region.

Today Clarity offers a footer region, to place any content. However getting column layout would not be a free service offered. The client would have to re-invent a UI solution that mimicked column-aligned cells.

## Examples

Working demo with Clarity 2. Resize the grid and columns to see the visible, fluid alignment of custom summary row projected into the footer region. It will naturally flow and scroll horizontally as necessary.

http://mattmutt.github.io/ngx-card-deck-seed/#/dashboard/dashboard-page/com_company_sample2/deck1

<img width="542" alt="customized-footer-column-alignment-zone-demo" src="https://user-images.githubusercontent.com/8688570/61889835-022afd80-aebb-11e9-8236-0bc14285f552.png">

## Workarounds

Each user has to figure out the math, CSS of the internal structure and alignment behavior that Clarity has implemented. Afterwards, armed with that knowledge would have to redefined a holding container with cells that are perfectly sized up such that content would match the original column in the above datagrid body.

## Standard UX practices

![excel-footer](https://user-images.githubusercontent.com/8688570/61889847-07884800-aebb-11e9-94f9-8f5375edee5b.jpg)

![locked-footer-design](https://user-images.githubusercontent.com/8688570/61889865-1242dd00-aebb-11e9-894d-aba273dcc64f.png)

![fixed-footer-visual](https://user-images.githubusercontent.com/8688570/61889881-1d960880-aebb-11e9-92ac-ef3707bf39e6.png)

**Comment by hippee-lee** [comment]
Thanks for the contribution @mattmutt. This datagrid feature will need to go through the usual design/dev workflows as described in the contributing documents. I've marked this as an enhancement and put it into the backlog so that it can be prioritized accordingly.

**Comment by gnomeontherun** [comment]
The feature request here has been captured into our list and we’re going to take it into consideration as we develop Clarity Core capabilities. In an effort to clean up our backlog and focus our attention, I’m going to close this as captured in our feature requests. Please follow our development and releases to see when we release relevant components to make this possible. Future feature requests can be made in our [GitHub Discussions](https://github.com/vmware/clarity/discussions).
